rclcpp/include/rclcpp/service.hpp

```diff
@@ -88,13 +88,26 @@ class Service : public ServiceBase
   typedef std::function<
       void (const std::shared_ptr<typename ServiceT::Request> &,
       std::shared_ptr<typename ServiceT::Response> &)> CallbackType;
+
+  typedef std::function<
+      void (const std::shared_ptr<rmw_request_id_t> &,
+      const std::shared_ptr<typename ServiceT::Request> &,
+      std::shared_ptr<typename ServiceT::Response> &)> CallbackWithHeaderType;
```

Comment:
The wrapping looks a bit unnatural. Would it be better to add a newline after `void(` and indent all arguments one level?